### PR TITLE
Require Sorting by Slug

### LINKED ISSUE #5098: Require brands of the same name to be sorted by slug
Currently [Handshake (Protocol) is listed before Handshake](https://github.com/simple-icons/simple-icons/blob/f464deebc2e6ecfe8da62fac3399a25141cb56f9/_data/simple-icons.json#L3432-L3442) but [Hive (Blockchain) is listed after Hive](https://github.com/simple-icons/simple-icons/blob/f464deebc2e6ecfe8da62fac3399a25141cb56f9/_data/simple-icons.json#L3558-L3568) in `simple-icons.json`.

For consistency, I suggest that we add a step to our linting process to ensure that entries with the same `title` are sorted by their `slug` with the one that doesn't have a `slug` coming first and the remainder sorted lexicographically.

### PR DESCRIPTION
**Pull request by PeterShaggyNoble**
**Issue:** Fixes #5098
**Alexa rank:** n/a

### Checklist
  - [x] I updated the JSON data in `_data/simple-icons.json`
  - [ ] I optimized the icon with SVGO or SVGOMG
  - [ ] The SVG `viewbox` is `0 0 24 24`

### Description
This PR adds a step to our linting that requires icons in our JSON with the same `title` to be sorted by their `slug` entry with the one without a `slug` coming first and the rest sorted alphabetically. If I've done this right, then the initial commit should fail on the 2 "Handshake" icons but, as I don't have a local setup to properly test with, I could really do with someone else throwing an eye over it.

Also, do we think it's enough to just log the `title` when icons aren't sorted by `slug` or should we somehow display the `slug` as well? One way we could do it would be to replace the whole `alphabetical` function with the following:
```
  alphabetical: function() {
    const collector = (invalidEntries, icon, index, array) => {
      if (index > 0) {
        const prev = array[index - 1];
        const compare = icon.title.localeCompare(prev.title);
        if (compare < 0) {
          invalidEntries.push(icon.title);
        }else if(compare === 0){
          if(prev.slug){
            if(!icon.slug){
              invalidEntries.push(icon.title);
            }else if(icon.slug.localeCompare(prev.slug) < 0){
              invalidEntries.push(icon.title+` (${icon.slug})`);
            }
          }
        }
      }
      return invalidEntries;
    };

    const invalids = data.icons.reduce(collector, []);
    if (invalids.length) {
      return `Some icons aren't in alphabetical order:
        ${invalids.join(", ")}`;
    }
  }
```

### CODE REVIEW

File: scripts/lint.js [hunk 1]
```diff
@@ -25,8 +25,17 @@ const TESTS = {
...
+        if (compare < 0) {
           invalidEntries.push(icon);
+        }else if(compare === 0){
+          if(prev.slug){
+            if(!icon.slug){
+              invalidEntries.push(icon);
+            }else if(icon.slug.localeCompare(prev.slug) < 0){
+              invalidEntries.push(icon);
+            }
+          }
```
Comment (ericcornelissen) [inline scripts/lint.js:38]:
```suggestion
        } else if (compare === 0) {
          if (prev.slug) {
            if (!icon.slug || icon.slug.localeCompare(prev.slug) < 0) {
              invalidEntries.push(icon);
            }
          }
```

**Review by ericcornelissen — CHANGES_REQUESTED** [review]
Apart from the formatting this looks good to me :+1: 

And as for:

> Also, do we think it's enough to just log the `title` when icons aren't sorted by `slug` or should we somehow display the `slug` as well? One way we could do it would be to replace the whole `alphabetical` function with the following:
> 
> ```
>   alphabetical: function() {
>     const collector = (invalidEntries, icon, index, array) => {
>       if (index > 0) {
>         const prev = array[index - 1];
>         const compare = icon.title.localeCompare(prev.title);
>         if (compare < 0) {
>           invalidEntries.push(icon.title);
>         }else if(compare === 0){
>           if(prev.slug){
>             if(!icon.slug){
>               invalidEntries.push(icon.title);
>             }else if(icon.slug.localeCompare(prev.slug) < 0){
>               invalidEntries.push(icon.title+` (${icon.slug})`);
>             }
>           }
>         }
>       }
>       return invalidEntries;
>     };
> 
>     const invalids = data.icons.reduce(collector, []);
>     if (invalids.length) {
>       return `Some icons aren't in alphabetical order:
>         ${invalids.join(", ")}`;
>     }
>   }
> ```

I think that's a thing we should look into. However, I would rather change [this line](https://github.com/simple-icons/simple-icons/blob/1dcec241332b77b7c8a4dd3e18835c6eafb878ac/scripts/lint.js#L38) (e.g. with a helper) function for formatting the string than doing that in the `collector` function. E.g. something along the lines of

```js
const collector = (invalidEntries, icon, index, array) => { ... };
const format = (icon) => {
  if (icon.slug) {
    return `${icon.title} (${icon.slug})`;
  }

  return icon.title
};

const invalids = data.icons.reduce(collector, []);
if (invalids.length) {
  return `Some icons aren't in alphabetical order:
    ${invalids.map(icon => format(icon)).join(", ")}`;
}
```

**Comment by PeterShaggyNoble** [comment]
Thanks, @ericcornelissen, formatting updated. And I agree with your suggested change for including the slug in the error; will get that taken care of later on today.

**Comment by PeterShaggyNoble** [comment]
Pushed the update to include the slug in the error log.

File: scripts/lint.js [hunk 2]
```diff
@@ -25,17 +25,30 @@ const TESTS = {
...
         }
       }
       return invalidEntries;
     };
+    const format = icon => {
+            if (icon.slug) {
+                return `${icon.title} (${icon.slug})`;
+            }
+            return icon.title;
+    };
```
Comment (ericcornelissen) [inline scripts/lint.js:46]:
```suggestion
    const format = icon => {
      if (icon.slug) {
        return `${icon.title} (${icon.slug})`;
      }
      return icon.title;
    };
```

**Review by ericcornelissen — CHANGES_REQUESTED** [review]
Again, part from the formatting (:sweat_smile:) this looks good to me :+1:

File: scripts/lint.js (hunk 2 above)
Comment (PeterShaggyNoble) [reply scripts/lint.js:46]: Won't be the last time I do that! 😆 I use tabs for indentation almost exclusively; easy to make mistakes when using spaces instead.

**Review by ericcornelissen — APPROVED** [review]

**Comment by ericcornelissen** [comment]
Apologies, messed up a bit but this is merged with 6c435fe7411182ea8e09c42d77ba827e37bb79f0 :sweat_smile:

**Pull request closed without merging**